pkgs/common-updater/scripts/list-directory-versions

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+
+import argparse
+import requests
+import os
+import subprocess
+import re
+from bs4 import BeautifulSoup
+
+parser = argparse.ArgumentParser(description="lists all available versions listed for a package in a site.")
+
+os.environ.get('UPDATE_NIX_PNAME', '')
+
+parser.add_argument("--pname", default=os.environ.get('UPDATE_NIX_PNAME', ''), help="Name of the package.")
+parser.add_argument("--attr-path", default=os.environ.get('UPDATE_NIX_ATTR_PATH', ''), help="Attribute path of the package.")
+parser.add_argument("--url", help="Url of the page that lists the package versions.")
+parser.add_argument("--file", help="File name for writing debugging information.")
+
+if __name__ == "__main__":
+    args = parser.parse_args()
+
+    pname = getattr(args, "pname")
+    attr_path = getattr(args, "attr_path")
+    url = getattr(args, "url")
+    file = getattr(args, "file")
+
+    if not url:
+        url=subprocess.check_output(
+            ["nix-instantiate", "--eval", "-E",
+             f"with import ./. {{}}; dirOf (lib.head {attr_path}.src.urls)",
+             ],
+            text=True,
+        ).strip('"\n')
```

Comment:
Not sure if Nix might not escape some special character when printing. Would recommend passing `--json` to `nix-instantiate` and then using `json.loads` instead of stripping `"`.

Reply:
Done